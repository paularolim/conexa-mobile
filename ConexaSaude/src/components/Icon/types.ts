export type IconName = 'calendar' | 'house' | 'plus' | 'user' | 'eye-on' | 'eye-off';

export interface IconProps {
  name: IconName;
  width?: number;
  height?: number;
  fill?: string;
}
