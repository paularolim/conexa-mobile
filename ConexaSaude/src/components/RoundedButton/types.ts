export type { TouchableOpacityProps as RoundedButtonProps } from 'react-native';
