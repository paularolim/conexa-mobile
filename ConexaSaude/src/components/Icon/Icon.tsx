import Calendar from '@assets/icons/calendar.svg';
import EyeOff from '@assets/icons/eye-off.svg';
import EyeOn from '@assets/icons/eye-on.svg';
import House from '@assets/icons/house.svg';
import Plus from '@assets/icons/plus.svg';
import SignOut from '@assets/icons/sign-out.svg';
import User from '@assets/icons/user.svg';

import { IconProps } from './types';

export function Icon({ name, width = 40, height = 40, fill = 'none' }: IconProps) {
  switch (name) {
    case 'calendar':
      return <Calendar width={width} height={height} stroke={fill} />;
    case 'house':
      return <House width={width} height={height} stroke={fill} />;
    case 'plus':
      return <Plus width={width} height={height} fill={fill} />;
    case 'user':
      return <User width={width} height={height} stroke={fill} />;
    case 'eye-off':
      return <EyeOff width={width} height={height} stroke={fill} />;
    case 'eye-on':
      return <EyeOn width={width} height={height} stroke={fill} />;
    case 'sign-out':
      return <SignOut width={width} height={height} stroke={fill} />;
    default:
      throw new Error('Icon name is required');
  }
}
