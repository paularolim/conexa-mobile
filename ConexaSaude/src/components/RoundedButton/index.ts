export * from './RoundedButton';
export * from './types';
