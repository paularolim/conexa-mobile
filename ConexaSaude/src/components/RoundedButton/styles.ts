import { StyleSheet, TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';

export const Container = styled(TouchableOpacity)`
  width: 70px;
  height: 50px;
  background-color: ${({ theme }) => theme.colors.pink};
  border-radius: 12px;
  top: 5px;
`;

export const styles = StyleSheet.create({
  shadow: {
    shadowColor: '#f28080',
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 1,
    shadowRadius: 5,
    elevation: 50,
  },
});
