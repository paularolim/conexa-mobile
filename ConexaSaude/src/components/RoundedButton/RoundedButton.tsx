import { Container, styles } from './styles';
import { RoundedButtonProps } from './types';

export function RoundedButton({ children, onPress, style, ...rest }: RoundedButtonProps) {
  return (
    <Container activeOpacity={1} onPress={onPress} {...rest} style={[style, { ...styles.shadow }]}>
      {children}
    </Container>
  );
}
