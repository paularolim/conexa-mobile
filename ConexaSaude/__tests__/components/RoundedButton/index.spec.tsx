import renderer from 'react-test-renderer';

import { Icon } from '@components/Icon';
import { RoundedButton } from '@components/RoundedButton';

import { Wrapper } from '../../Wrapper';

describe('RoundedButton component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <RoundedButton>
            <Icon name="plus" width={32} height={32} />
          </RoundedButton>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
