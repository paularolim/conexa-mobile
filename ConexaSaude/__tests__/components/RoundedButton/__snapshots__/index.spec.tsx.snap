// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton component should render default component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#F28080",
      "borderRadius": 12,
      "elevation": 50,
      "height": 50,
      "opacity": 1,
      "shadowColor": "#f28080",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": 70,
    }
  }
>
  <svg
    data-file-name="SvgPlus"
    fill="none"
    height={32}
    width={32}
  />
</View>
`;
